### PR TITLE
feat(cli): add global sync override option

### PR DESCRIPTION
**Pull request by zachferland**
Follow up to - https://github.com/ceramicnetwork/js-ceramic/pull/1498

Adds sync override flag, as options to globally force sync mode on all requests. Required by gateway.

### CODE REVIEW

File: packages/core/src/ceramic.ts [hunk 1]
```diff
@@ -121,8 +122,15 @@ export interface CeramicParameters {
   gateway: boolean,
   networkOptions: CeramicNetworkOptions,
   validateStreams: boolean,
+  syncOverride: SyncConfig
 }
 
+/**
+ * Global Sync Config Options
+ */
+export interface SyncConfig {
```
Comment (stbrody) [inline packages/core/src/ceramic.ts:131]: why do we need this extra interface?  Couldn't `syncOverride` just be a `SyncOptions`?

Comment (stbrody) [reply packages/core/src/ceramic.ts:131]: If you're trying to future-proof to a world where we may want to add other options to the override, such as `syncTimeoutSeconds` for example, you could have this be a `LoadOpts` and call the field `loadOptsOverride`

**Review by stbrody — COMMENTED** [review]
looks good, one question about the new interface

File: packages/core/src/ceramic.ts (hunk 1 above)
Comment (zachferland) [reply packages/core/src/ceramic.ts]: Yeah was leaning towards that, just change/updated. Basically imagine options as something like this after
```
opts = { ...DEFAULT_CREATE_FROM_GENESIS_OPTS, this._loadOptsDefault, ...opts, ...this._loadOptsOverride}
```

**Review by stbrody — APPROVED** [review]
LGTM

**Comment by zachferland** [comment]
Not sure whats up with tests right now, forgot to merge last week, merged/fixed with dev again, @ukstv or @stbrody please merge if tests are fine

**Comment by ukstv** [comment]
@zachferland rebased on the current prettified develop in https://github.com/ceramicnetwork/js-ceramic/pull/1541

**Pull request closed without merging**